contracts/src/deployers/savings-usds-l2/SavingsUSDSL2HyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,171 @@
+    function _prepareInitialize(
+        IHyperdrive _hyperdrive,
+        address _lp,
+        uint256 _contribution,
+        IHyperdrive.Options memory _options
+    ) internal override returns (uint256 value) {
+        // If base is the deposit asset, the initialization will be paid in the
+        // base token.
+        address token;
+        if (_options.asBase) {
+            token = _hyperdrive.baseToken();
+        }
+        // Otherwise, the initialization will be paid in vault shares.
+        else {
+            token = _hyperdrive.vaultSharesToken();
+        }
+
+        // Take custody of the contribution and approve Hyperdrive to pull the
+        // tokens.
+        ERC20(token).safeTransferFrom(_lp, address(this), _contribution);
+        ERC20(token).forceApprove(address(_hyperdrive), _contribution);
+
+        return value;
+    }
+
+    /// @notice Convert an amount of vault shares to an amount of base.
+    /// @param _shareAmount The vault shares amount.
+    /// @return The base amount.
+    function convertToBase(uint256 _shareAmount) public view returns (uint256) {
+        return SavingsUSDSL2Conversions.convertToBase(PSM, _shareAmount);
+    }
+
+    /// @notice Convert an amount of base to an amount of vault shares.
+    /// @param _baseAmount The base amount.
+    /// @return The vault shares amount.
+    function convertToShares(
+        uint256 _baseAmount
+    ) public view returns (uint256) {
+        return SavingsUSDSL2Conversions.convertToShares(PSM, _baseAmount);
+    }
+
+    /// @dev We override the message value check since this integration is
+    ///      not payable.
+    function _checkMessageValue() internal view override {
+        if (msg.value != 0) {
+            revert IHyperdriveDeployerCoordinator.NotPayable();
+        }
+    }
+
+    /// @notice Checks the pool configuration to ensure that it is valid.
+    /// @param _deployConfig The deploy configuration of the Hyperdrive pool.
+    /// @param _extraData The empty extra data.
+    function _checkPoolConfig(
+        IHyperdrive.PoolDeployConfig memory _deployConfig,
+        bytes memory _extraData
+    ) internal view override {
+        // Perform the default checks.
+        super._checkPoolConfig(_deployConfig, _extraData);
+
+        // Ensure that the vault shares token address is properly configured.
+        if (address(_deployConfig.vaultSharesToken) != address(PSM.susds())) {
+            revert IHyperdriveDeployerCoordinator.InvalidVaultSharesToken();
+        }
+
+        // Ensure that the base token address is properly configured.
+        if (address(_deployConfig.baseToken) != address(PSM.usds())) {
+            revert IHyperdriveDeployerCoordinator.InvalidBaseToken();
+        }
+
+        // Ensure that the minimum share reserves are equal to 1e15. This value
+        // has been tested to prevent arithmetic overflows in the
+        // `_updateLiquidity` function when the share reserves are as high as
```

Comment:
This comment is outdated. It can support much larger share reserves

Reply:
Deleted the part about 200M shares.